url/src/lib.rs

```diff
@@ -1751,14 +1783,28 @@ impl Url {
                 }
                 parser.parse_cannot_be_a_base_path(parser::Input::new_no_trim(path));
             } else {
-                let mut has_host = true; // FIXME
                 parser.parse_path_start(
                     scheme_type,
                     &mut has_host,
                     parser::Input::new_no_trim(path),
                 );
             }
         });
+
+        // For cases where normalization is applied across both the serialization and the path.
+        // Append "/." immediately after the scheme (up to ":")
+        // This is done if three conditions are met.
+        // https://url.spec.whatwg.org/#url-serializing
+        // 1. The host is null
+        // 2. The url's path length is greater than 1
+        // 3. the first segment of the URL's path is an empty string
+        if !has_host && path.len() > 1 && path_empty {
```

Comment:
Shouldn't the path_empty check apply to the new path that we just parsed in the block above?

Reply:
Nevermind, The path empty bit applies to the first one, meaning if we have had a `/.` before setting the path, we needed to add back the `/.` (empty path) back afterwards.